app/src/main/java/io/homeassistant/companion/android/settings/ssid/SsidDialogFragment.kt

```diff
@@ -0,0 +1,97 @@
+package io.homeassistant.companion.android.settings.ssid
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import androidx.lifecycle.lifecycleScope
+import androidx.preference.PreferenceDialogFragmentCompat
+import io.homeassistant.companion.android.DaggerPresenterComponent
+import io.homeassistant.companion.android.PresenterModule
+import io.homeassistant.companion.android.R
+import io.homeassistant.companion.android.common.dagger.GraphComponentAccessor
+import io.homeassistant.companion.android.databinding.DialogSsidBinding
+import io.homeassistant.companion.android.domain.url.UrlUseCase
+import javax.inject.Inject
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class SsidDialogFragment : PreferenceDialogFragmentCompat() {
+
+    companion object {
+        fun newInstance(preferenceKey: String): SsidDialogFragment = SsidDialogFragment().apply {
+            arguments = Bundle().apply { putString(ARG_KEY, preferenceKey) }
+        }
+    }
+
+    @Inject lateinit var urlUseCase: UrlUseCase
+
+    private val ssidAdapter = SsidRecyclerViewAdapter()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        DaggerPresenterComponent
+            .builder()
+            .appComponent((requireContext().applicationContext as GraphComponentAccessor).appComponent)
+            .presenterModule(PresenterModule(this))
+            .build()
+            .inject(this)
```

Comment:
The dialog fragment shouldn't be using dagger to load anything.  It should all be passed into the preference.